packages/@aws-cdk/aws-apigateway/lib/resource.ts

```diff
@@ -179,6 +179,11 @@ export class ProxyResource extends Resource {
       defaultMethodOptions: props.defaultMethodOptions,
     });
 
+    // Add ANY method to parent but only if it's the root
```

Comment:
Didn't we say we want this behavior in an override of `addMethod`, so it will apply to methods added manually as well?